core/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -103,6 +103,7 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
   private val stateLock = new ReentrantLock
 
   val extraCoresPerExecutor = conf.getInt("spark.mesos.extra.cores", 0)
+  val maxGpus = conf.getInt("spark.mesos.gpus.max", 0)
```

Comment:
I know in a previous iteration of this patch we were using the config variable `spark.mesos.gpu.enabled=true` instead of `spark.mesos.gpu.max=XX`. Why did this change? If I nee to set a max, how do I know how big to make the value? Do the spark configuration scripts allow me to run bash commands to programattically count the total number of GPUs on the machine if that's my desired max?